crates/sui-types/src/deny_list.rs

```diff
@@ -43,6 +46,73 @@ pub struct PerTypeDenyList {
     pub denied_addresses: Table,
 }
 
+impl DenyList {
+    pub fn check_coin_deny_list(
+        address: SuiAddress,
+        coin_types: BTreeSet<String>,
+        object_store: &dyn ObjectStore,
+    ) -> UserInputResult {
+        let Some(deny_list) = get_coin_deny_list(object_store) else {
+            // TODO: This is where we should fire an invariant violation metric.
+            if cfg!(debug_assertions) {
+                panic!("Failed to get the coin deny list");
+            } else {
+                return Ok(());
+            }
+        };
+        Self::check_deny_list(deny_list, address, coin_types, object_store)
+    }
+
+    fn check_deny_list(
+        deny_list: PerTypeDenyList,
+        address: SuiAddress,
+        coin_types: BTreeSet<String>,
+        object_store: &dyn ObjectStore,
+    ) -> UserInputResult {
+        // TODO: Add caches to avoid repeated DF reads.
+        let Ok(count) = get_dynamic_field_from_store::<SuiAddress, u64>(
+            object_store,
+            deny_list.denied_count.id,
+            &address,
+        ) else {
+            return Ok(());
+        };
+        if count == 0 {
+            return Ok(());
+        }
+        for coin_type in coin_types {
+            let Ok(denied_addresses) = get_dynamic_field_from_store::<Vec<u8>, VecSet<SuiAddress>>(
+                object_store,
+                deny_list.denied_addresses.id,
+                &coin_type.clone().into_bytes(),
```

Comment:
I think this needs to be the bcs bytes? And if so we should update the name/comment on get_dynamic_field_from_store to reflect that

Reply:
It does not. Inside the function it will bcs serialize it.